File: VB/DetailGridSearchExample/ViewModel/TaskViewModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace DetailGridSearchExample
	Friend Class TaskViewModel
		Private _List As ObservableCollection(Of TaskList)

		Public ReadOnly Property List() As ObservableCollection(Of TaskList)
			Get
				If _List Is Nothing Then
					_List = New ObservableCollection(Of TaskList)()
					For i As Integer = 0 To 9
						Dim taskList As New TaskList() With {
							.TaskGroup = "Group" & i,
							.GroupNumber = i,
							.List = New ObservableCollection(Of Task)()
						}
						For j As Integer = 0 To 999
							taskList.List.Add(New Task() With {
								.Name = "Task" & i & "." & j,
								.Date = New Date(2014, 10, (New Random()).Next(1, 31)),
								.IsCompleted = j Mod 2 <> 0
							})
						Next j
						_List.Add(taskList)
					Next i
				End If
				Return _List
			End Get
		End Property

		Public Class Task
			Public Property Name() As String
			Public Property [Date]() As Date
			Public Property IsCompleted() As Boolean
		End Class

		Public Class TaskList
			Public Property TaskGroup() As String
			Public Property GroupNumber() As Integer
			Public Property List() As ObservableCollection(Of Task)
		End Class
	End Class
End Namespace

File: CS/DetailGridSearchExample/TaskViewModel.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DetailGridSearchExample
{
    class TaskViewModel
    {
        private ObservableCollection<TaskList> _List;

        public ObservableCollection<TaskList> List
        {
            get
            {
                if (_List == null)
                {
                    _List = new ObservableCollection<TaskList>();
                    for (int i = 0; i < 10; i++)
                    {
                        TaskList taskList = new TaskList() { TaskGroup = "Group" + i, GroupNumber = i, List = new ObservableCollection<Task>() };
                        for (int j = 0; j < 1000; j++)
                        {
                            taskList.List.Add(new Task() { Name = "Task" + j, Date = new DateTime(2014, 10, new Random().Next(1, 31)), IsCompleted = j % 2 != 0 });
                        }
                        _List.Add(taskList);
                    }
                }
                return _List;
            }
        }

        public class Task
        {
            public string Name { get; set; }
            public DateTime Date { get; set; }
            public bool IsCompleted { get; set; }
        }

        public class TaskList
        {
            public string TaskGroup { get; set; }
            public int GroupNumber { get; set; }
            public ObservableCollection<Task> List { get; set; }
        }
    }
}
